be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1505,14 +1562,16 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
 
     _current_batch_rows_read = 0;
     uint32_t nrows_read_limit = _opts.block_row_max;
-    if (UNLIKELY(_estimate_row_size)) {
-        // read 100 rows to estimate average row size
+    if (UNLIKELY(_opts.stats->blocks_load == 0)) {
```

Comment:
Not depend on the stats, add another variable to indicate that this is first read.

Reply:
fixed
// Read up to 100 rows at a time while waiting for the estimated row size.